langchain/document_loaders/alibaba_cloud_maxcompute.py

```diff
@@ -0,0 +1,85 @@
+from typing import List, Optional
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class MaxComputeLoader(BaseLoader):
+    """Loads a query result from MaxCompute table into a list of documents.
+
+    Each document represents one row of the result. The `page_content_columns`
+    are written into the `page_content` of the document. The `metadata_columns`
+    are written into the `metadata` of the document. By default, all columns
+    are written into the `page_content` and none into the `metadata`.
+
+    `page_content_columns` represents columns in the MaxCompute table, the
+    data of these columns will be stored in the database for future queries
+    or matching.
+    `meta_data` indicates certain columns in the MaxCompute table, and the
+    data of these columns are displayed with the `page_content` to provide
+    original meta information.
+    """
+
+    def __init__(
+        self,
+        query: str,
+        accesskey_id: str,
+        accesskey_secret: str,
+        endpoint: str,
+        project: str,
+        page_content_columns: Optional[List[str]] = None,
+        metadata_columns: Optional[List[str]] = None,
+    ):
+        self.query = query
+        self.accesskey_id = accesskey_id
+        self.accesskey_secret = accesskey_secret
+        self.endpoint = endpoint
+        self.project = project
+        self.page_content_columns = page_content_columns
+        self.metadata_columns = metadata_columns
+
+        try:
+            from odps import ODPS
+        except ImportError as ex:
+            raise ValueError(
+                "Could not import pyodps python package. "
+                "Please install it with `pip install pyodps` or refer to "
+                "https://pyodps.readthedocs.io/."
+            ) from ex
+
+        self.odps = ODPS(
+            self.accesskey_id,
+            self.accesskey_secret,
+            self.project,
+            endpoint=self.endpoint,
+        )
+
+        assert self.odps.exist_project(
+            self.project
+        ), f"The project <{self.project}> does not exist."
+
+    def load(self) -> List[Document]:
+        docs: List[Document] = []
+
+        # Use SQL
+        with self.odps.execute_sql(self.query).open_reader() as reader:
+            assert (
+                reader.count >= 1
+            ), "Make sure at least one piece of data in the table."
```

Comment:
are we sure we want to raise an error if count == 0? could the table have data but given the query none was returned?